sql/src/main/java/org/apache/druid/sql/calcite/schema/SegmentMetadataHolder.java

```diff
@@ -105,14 +112,14 @@ public Builder(
         long isPublished,
         long isAvailable,
         long isRealtime,
-        long numReplicas
+        Map<String, Set<String>> segmentServerMap
     )
     {
       this.segmentId = segmentId;
       this.isPublished = isPublished;
       this.isAvailable = isAvailable;
       this.isRealtime = isRealtime;
-      this.numReplicas = numReplicas;
+      this.segmentServerMap = segmentServerMap;
```

Comment:
yeah, it should be immutable, made it so across the classes

Reply:
I think it would have been ok if variables and functions were left typed just plain `Map<String, Set<String>>`, and just assigned them an immutable copy, similar to what `RowSignature` is doing, though I suppose there is nothing wrong either way. 🙃

Reply:
Yeah, i knew something looked odd